mission/joystick_interface_auv/joystick_interface_auv/joystick_interface_auv.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/python3
+import rclpy
+from rclpy.node import Node
+from geometry_msgs.msg import Wrench
+from sensor_msgs.msg import Joy
+from std_msgs.msg import Bool
+
+
+class States:
+    XBOX_MODE = 1
+    AUTONOMOUS_MODE = 2
+    NO_GO = 3
+
+
+class JoystickInterface(Node):
+
+    def __init__(self):
+        super().__init__('joystick_interface_node')
+        self.get_logger().info(
+            "Joystick interface is up and running. \n When the XBOX controller is connected, press the killswitch button once to enter XBOX mode."
+        )
+
+        self.last_button_press_time_ = 0
+        self.debounce_duration_ = 0.25
+        self.state_ = States.NO_GO
+
+        self.joystick_buttons_map_ = [
+            "A",
+            "B",
+            "X",
+            "Y",
+            "LB",
+            "RB",
+            "back",
+            "start",
+            "power",
+            "stick_button_left",
+            "stick_button_right",
+        ]
+
+        self.joystick_axes_map_ = [
+            "horizontal_axis_left_stick",  #Translation (Left and Right)
+            "vertical_axis_left_stick",  #Translation (Forwards and Backwards)
+            "LT",  #Negative thrust/torque multiplier
+            "horizontal_axis_right_stick",  #Rotation
```

Comment:
Rotation? around what axis? Try to use better descriptors (Surge, Sway, Heave, Roll, Pitch, Yaw) for all of these